gateway/rudder-sources/api/api_test.go

```diff
@@ -0,0 +1,194 @@
+package api_test
+
+import (
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/http/httptest"
+	"regexp"
+	"testing"
+
+	gomock "github.com/golang/mock/gomock"
+	"github.com/rudderlabs/rudder-server/gateway/rudder-sources/api"
+	"github.com/rudderlabs/rudder-server/gateway/rudder-sources/model"
+	"github.com/stretchr/testify/require"
+)
+
+func TestDelete(t *testing.T) {
```

Comment:
can we have tests for the exceptional scenarios as well, e.g. when the service returns an error?

Reply:
added.